src/app/components/Header/Header.js

```diff
@@ -1,36 +1,57 @@
 import React from 'react'
 import PropTypes from 'prop-types'
+import { withRouter } from 'react-router'
 
 import NetworkSwitcher from '../NetworkSwitcher'
 import MainNav from '../MainNav'
 
-import './Header.css'
+import home from '../../../img/home.svg'
 
-const Header = (props) => {
-  const { showMenu, setNetwork, selectedNetworkId, networks } = props
+import style from './Header.css'
+
+const getNavigation = props => {
+  const { account, history } = props
+  const loggedIn = account.address && account.wif
+  const isHomepage = history.location.pathname === '/'
+
+  if (!loggedIn && isHomepage) {
+    return
+  }
+
+  let navigation
+
+  if (loggedIn) {
+    navigation = <MainNav />
+  } else {
+    navigation = (
+      <button className={ style.mainNavigationNotLoggedIn } onClick={ () => history.push('/') }>
+        <img className={ style.mainNavigationNotLoggedInImg } src={ home } alt='house' />
+      </button>
+    )
+  }
+
+  return navigation
+}
+
+const Header = props => {
```

Comment:
It's generally advised to have one component per file.

Reply:
Sure. This getNavigation isn't exactly a component, just a helper function to abstract some of the logic from the render function. Extract it to it's own file?